lib/SILOptimizer/Utils/OwnershipOptUtils.cpp

```diff
@@ -343,6 +364,59 @@ OwnershipLifetimeExtender::createPlusOneCopy(SILValue value,
   return result;
 }
 
+BeginBorrowInst *
+OwnershipLifetimeExtender::createPlusOneBorrow(SILValue value,
```

Comment:
For now, I think you need to assert/FIXME that the single use is not a reborrow. I have a utility for dealing with that.

Reply:
I see your point unless we eliminated the reborrow after this happened.